### ISSUE TITLE
cannot use ReactSelector and withAttribute()

### ISSUE THREAD

**Issue by mulyoved** [DevExpress/testcafe-react-selectors#14]
Trying to call 

```ReactSelector("AutoComplete").withAttribute("name", "city")``` 

does not work, get error **The specified selector does not match any element in the DOM tree.**

```Selector("input").withAttribute("name", "city")``` 

work fine

am I missing something?

**Comment by AlexanderMoskovkin** [comment]
Hi,

`ReactSelector('AutoComplete')` returns a root node of the component's render. Could you provide the structure of the `AutoComplete` component's render? Is the input element with `name=city` attribute is a root node of it?
If this input is a child of the component's root node you can use the following:
`ReactSelector("AutoComplete input").withAttribute("name", "city")`
or
`ReactSelector("AutoComplete").find('input').withAttribute("name", "city")`

**Comment by mulyoved** [comment]
@AlexanderMoskovkin you are right, adding input made it work

I was confused by the AutoComplete react props name="city" which is forwarded to the input as attribute name="city"

which raise I think nice feature request, be able to do something like that will be cool

```ReactSelector("AutoComplete").withProps("name", "city")```

**Comment by AlexanderMoskovkin** [comment]
Nice point! 
We should thoroughly think out the API for that to clearly separate html props (like `withAttribute`) from react props (like `props`, `state`)

**Comment by dimaip** [comment]
Just came here to open the same feature request: would be really nice to have that `withProps` selector.
Usecase: we have a set of presentational components that is used all over the page for different things (e.g. think of a Bootstrap-styled Button component). Now I would love to find some particular button based on its props, but currently that's not possible.

**Comment by jacobwgillespie** [comment]
Does implementing `withProps` require any changes to TestCafe itself or just this React selector?  I have a modified version of the selector that finds components by prop (via `ReactSelector('ComponentName', {prop: 'value'})`), but I'm not familiar enough with TestCafe to know if that code could be modified to fit the `withProps` style.  As in, I don't know if selectors are chainable.

**Comment by kirovboris** [comment]
Hi @jacobwgillespie, 

the extending of `ReactSelector` requires some changes in TestCafe. It's necessary to chain calls like 
```js
const el       = ReactSelector(...).withProps(...);
const elSubset = el.withProps(...);
```
We're working on this functionality now and going to implement it soon.

**Comment by jacobwgillespie** [comment]
@kirovboris awesome, thanks for the clarification!  I'll keep using my simple fork until the official chaining support is available.  👍

**Comment by kirovboris** [comment]
Implemented in `2.0.0`
